app/commands.py

```diff
@@ -1411,6 +1376,31 @@ async def rmpriv(ctx: Context) -> Optional[str]:
     return f"Updated {t}'s privileges."
 
 
+@command(Privileges.DEVELOPER, hidden=True)
+async def givedonator(ctx: Context) -> Optional[str]:
+    """Gives donator to a specified player (by name) for a specified amount of time, such as '3h5m'."""
+    if len(ctx.args) < 2:
+        return "Invalid syntax: !givedonator <name> <duration>"
+
+    if not (t := await app.state.sessions.players.from_cache_or_sql(name=ctx.args[0])):
+        return "Could not find user."
+
+    if not (timespan := timeparse("".join(ctx.args[1:]))):
+        return "Invalid timespan."
+
+    if t.donor_end < time.time():
+        timespan += int(time.time())
+    else:
+        timespan += t.donor_end
+
+    await app.state.services.database.execute(
+        "UPDATE users SET donor_end = :end WHERE id = :user_id",
+        {"end": timespan, "user_id": t.id},
+    )
+
+    return f"Added {''.join(ctx.args[1:])} of donator status to {t}."
```

Comment:
should be just ctx.args[1]

Reply:
i mean people may use spaces so incase they do it'll still support it